formats/roam/roam_dl_attachment.ts

```diff
@@ -0,0 +1,51 @@
+import { ColorComponent, Vault } from 'obsidian';
+import axios from 'axios';
+import { fs, path } from 'filesystem';
+
+export async function downloadFirebaseFile(line: string, attachmentsFolder:string) {
+    try {
+        let link: RegExpMatchArray | null;
+        let syntaxLink: RegExpMatchArray | null;
+        const regex = /{{pdf:|{{\[\[pdf|{{\[\[audio|{{audio:|{{video:|{{\[\[video/;
+        const imageRegex = /https:\/\/firebasestorage(.*?)\?alt(.*?)\)/;
+        const binaryRegex = /https:\/\/firebasestorage(.*?)\?alt(.*?)/
+        const regexImageNoDescription = /\!\[\]\((.+?)\)/;
+        const regexImagewithDescription = /\!\[(.+?)\]\((.+?)\)/;
+        if (regex.test(line)) {
+            link = line.match(/https:\/\/firebasestorage(.*?)\?alt(.*?)\}/);
+            syntaxLink = line.match(/{{.*https:\/\/firebasestorage.*?alt=media&.*?(?=\s|$)/);
+            
+        } else if (imageRegex.test(line)) {
+            link = line.match(imageRegex);
+            syntaxLink = line.match(/!\[.*https:\/\/firebasestorage.*?alt=media&.*?(?=\s|$)/);
+        } else {
+            // I expect this to be a bare link which is typically a binary file
+            link = line.match(binaryRegex);
+            syntaxLink = line.match(/https:\/\/firebasestorage.*?alt=media&.*?(?=\s|$)/);
+        } 
+        
+        if (link) {
+            const firebaseShort = 'https://firebasestorage' + link[1];
+            const firebaseUrl = link[0].slice(0, -1);
+            const response = await axios.get(firebaseUrl, { responseType: 'arraybuffer' });
```

Comment:
Can't use axios. You can use fetch, and fallback to `requestUrl` from `obsidian`.